trezor-keyring-mv2.js

```diff
@@ -0,0 +1,49 @@
+const { default: TrezorConnect, DEVICE_EVENT } = require('trezor-connect');
+const {
+  TrezorKeyring,
+  KEYRING_TYPE,
+  TREZOR_CONNECT_MANIFEST,
+} = require('./trezor-keyring');
+
+class TrezorKeyringMv2 extends TrezorKeyring {
+  constructor(opts = {}) {
+    super(opts);
+  }
+
+  init() {
+    TrezorConnect.on(DEVICE_EVENT, (event) => {
+      if (event && event.payload && event.payload.features) {
+        this.model = event.payload.features.model;
+      }
+    });
+
+    return TrezorConnect.init({ manifest: TREZOR_CONNECT_MANIFEST });
+  }
+
+  dispose() {
+    // This removes the Trezor Connect iframe from the DOM
+    // This method is not well documented, but the code it calls can be seen
+    // here: https://github.com/trezor/connect/blob/dec4a56af8a65a6059fb5f63fa3c6690d2c37e00/src/js/iframe/builder.js#L181
+    TrezorConnect.dispose();
+    return Promise.resolve();
+  }
```

Comment:
It is worth mentioning I noticed that this method is only present for trezor and ledger keyring, but they go under different names (`dispose` and `destroy`).

Perhaps it would be a good idea to standardise the name so that we can call it if it exists. Same as we are doing with `init`.

This is where they are being called.
https://github.com/MetaMask/metamask-extension/blob/d21757c029794fb8004f2d7a36a52b1fe801c5bf/app/scripts/metamask-controller.js#L4391

Reply:
Agreed! We've used `destroy` for this sort of thing elsewhere, so maybe that makes sense as the standard name. We can do that in a separate PR.